components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -400,24 +431,26 @@ void BraveReferralsService::GetFirstRunTime() {
 #if defined(OS_ANDROID)
   // Android doesn't use a sentinel to track first run, so we use a
   // preference instead.
-  first_run_timestamp_ =
+  first_run_timestamp =
       pref_service_->GetTime(kReferralAndroidFirstRunTimestamp);
-  if (first_run_timestamp_.is_null()) {
-    first_run_timestamp_ = base::Time::Now();
+  if (first_run_timestamp.is_null()) {
+    first_run_timestamp = base::Time::Now();
     pref_service_->SetTime(kReferralAndroidFirstRunTimestamp,
-                           first_run_timestamp_);
+                           first_run_timestamp);
   }
-  PerformFinalizationChecks();
+  SetFirstRunTime(first_run_timestamp);
 #else
-  task_runner_->PostTask(
-      FROM_HERE, base::Bind(&BraveReferralsService::GetFirstRunTimeDesktop,
-                            base::Unretained(this)));
+  task_runner_->PostTaskAndReplyWithResult(
+      FROM_HERE, base::BindOnce(&first_run::GetFirstRunSentinelCreationTime),
```

Comment:
Is this just for tests? If so might be worth commenting. Sentinel gets read & cached in chrome main process

Reply:
Ah, it's already cached. so I think we can call in UI thread directly.
but not sure that causes threading failure during the test.

Reply:
No, there are two commits here. One for the test issue and one to fix unsafe the threading issues I found.

Reply:
@simonhong I just followed what was already there. It's possible this could be optimized, but I didn't want to change anything here other than make it safe